File: 2231-find-first-palindromic-string-in-the-array/2231-find-first-palindromic-string-in-the-array.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        i, j = 0, len(s) - 1
        while i < j:
            if s[i] != s[j]:
                return False
            i += 1
            j -= 1
        return True
    
    def firstPalindrome(self, words: list[str]) -> str:
        for word in words:
            if self.isPalindrome(word):
                return word
        return ""

File: 2231-find-first-palindromic-string-in-the-array/2231-find-first-palindromic-string-in-the-array.java
import java.util.ArrayList;
import java.util.List;

class Solution {
    public boolean isPalindrome(String s) {
        int i = 0;
        int j = s.length() - 1;
        while (i < j) {
            if (s.charAt(i) != s.charAt(j)) {
                return false;
            }
            i++;
            j--;
        }
        return true;
    }
    
    public String firstPalindrome(String[] words) {
        List<String> wordList = new ArrayList<>();
        for (String word : words) {
            wordList.add(word);
        }
        
        for (String word : wordList) {
            if (isPalindrome(word)) {
                return word;
            }
        }
        return "";
    }
}
